New Away Mission map

OceanFish1 opened the pull request:
## About The Pull Request

Hey hey people, Fish here. Today I'll be pulling a new away mission I was making for months. It's called "The Factory". So, this is the kind of mission where you supposed to get tons of gamer loot, but except one thing. You don't get it for free this time. You had too much for previous missions (except underground outpost which is usually used for plasmaman erotic role play and burning everyone alive). You're gonna struggle to get your gamer loot, but be sure it worth it. The map size is 225x210 and only one z-level. Why only one? Because we fuck multi-z level maps, we don't do that here. Instead we do tons teleportation to make the player feel they're actually going up or down.
Now the difficulty of the mission. This contains good amount of mobs, where most of them can shoot. Some of them can shoot your limb or your brains off and two of them may husk you instantly. The good thing is that they're all lootable, which means when the die they don't magically turn into human gibs. They leave bodies with all the things they had when they were alive.
Now about the plot. 
Who the fuck needs a plot for an away mission? You just go in and do whatever. I may already broke your aesthetic feelins, so here's your plot: A group of people on Earth decided to make a company that specializes on finding ways to produce infinity food. They found some god's forgotten village and built a big building. They decided that villagers should not know what's happening inside and made the outer walls look like it's made out of wood. Then they built an underground facility. Then they went too far and built a gateway, the same like on the station. They made a special squad for gateway expeditions that is tasked to go in, grab whatever they see, get bonus points if it's behaving way too strange for people and bring it to factory for """""research""""". They were stealing shit from wizards, nanotrasen, other people. Your mission goal is to go """""""talk""""""" with them and ask them to stop stealing. All you need to do for this is to travel in the past, when they didn't abandon the place. Yes, you gotta time travel and """"""ask"""""" their boss to turn off the gates. Sounds easy.

Since it's the first PR I ever made and the first attempt to code I won't be surprized if I already commited several warcrimes while was writing this. But at least I found myself as a (good) mapper. I even got to do some sprites. Yes, that pile of wood is actually a shotgun. Most of the sounds are already existing sounds, but edited. Some of them are mine.
 If things went wrong please tell me about them. Especially about the balance.

![factory](https://user-images.githubusercontent.com/65843722/102172394-3f869580-3ecb-11eb-9c72-96fc091f9d73.png)

## Why It's Good For The Game

The previous missions are outdated as fuck and thats why they're being reworked. But while it's not done yet, I wanna pull my own mission because I don't see a reason why not.

## Changelog
:cl:
add: New away mission
add: New mobs and items for new away mission
soundadd: added few sounds for away mission mobs and ambience
imageadd: added icons for a lever-action shotgun, tactical sunglasses and some mobs
code: made gatling laser emaggable
config: added The Factory to away missions
/:cl:

OceanFish1 committed:
test

123

OceanFish1 committed:
Revert "test"

This reverts commit f60596bcf97d486efc5a6ec2d3bcb33d676cf8ae.

OceanFish1 committed:
New Away Mission

new away mission

OceanFish1 committed:
Revert "New Away Mission"

This reverts commit 10255a4ec81de63e34565d2551b979c0486fd185.

OceanFish1 committed:
New Away Mission

i'm gum

OceanFish1 committed:
sounds remade

sounds good also added vomit niceeeee

Fronsis commented:
Looks very nice, you should check with Cae to have it added once his rework is finished

OceanFish1 commented:
> Looks very nice, you should check with Cae to have it added once his rework is finished

Yeah, we've been talking about that.

zeskorion commented:
ha ah ha ha hah aha hahahahahhahahahhhhhhhaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

PowerfulBacon commented:
When it comes to gamer loot, there is no such thing as hard to get gamer loot. There are always over 100 ways to cheese the threat in a way you haven't thought of.

OceanFish1 commented:
> When it comes to gamer loot, there is no such thing as hard to get gamer loot. There are always over 100 ways to cheese the threat in a way you haven't thought of.

Of course I know that, but first I need to see what in ways the threat will be avoided. There is also a thing that sometimes the threat IS the loot.

OceanFish1 committed:
did some mapping stuff

deleted useless paper and replaced mind batterer to fit with the plot

PowerfulBacon reviewed (dismissed):
Incorrect map format. Must use TGM

knacker48 commented:
The key isn't to make it uncheeseable, its to make it hard enough that you need to put in a lot of effort to cheese. If you have a geneticist with tele and xray attacking your mobs with a crowbar two rooms away, make your stronger guys teleport to your position or beeline to you, smashing walls along the way.

PowerfulBacon commented:
Don't include a_gifts

OceanFish1 committed:
yoo look i saved it in tgm format

hey lois i'm in the 2d spaceman  game now

OceanFish1 commented:
> Don't include a_gifts

also why no gifts? They may give something game-breaking or what?

PowerfulBacon commented:
> also why no gifts? They may give something game-breaking or what?

They can give any item in the game, including inaccessible things (Hence why only available on 1 day of the year for the memes).

OceanFish1 commented:
> > also why no gifts? They may give something game-breaking or what?
> 
> They can give any item in the game, including inaccessible things (Hence why only available on 1 day of the year for the memes).

understandable

OceanFish1 committed:
gifts are no more

christmas is ended

ItsTisme commented:
iam a sucker for factory aesthatic, would be nice if a away mision would be actuel production facility

OceanFish1 commented:
> iam a sucker for factory aesthatic, would be nice if a away mision would be actuel production facility

it has conveyor belts that are going circle isn't that enough?????///

OceanFish1 committed:
from turf to src

forgot to download counter strike source content

OceanFish1 committed:
fixed some var edit errors

poggers!

OceanFish1 committed:
RUN

NOW

Kryyto commented:
The "time travel" mechanic work well for me, the mob have some nice behavior (speaking and screaming new dialogues line for example). And they are indeed op, found it cool

OceanFish1 committed:
christmas even on bee

returning gifts with because christmas

OceanFish1 committed:
I DIDN'T CHECK THE CODE BEFORE PUTTING THOSE SWAT SHOES ON CORPSES

Stupid fucking mistakes man

OceanFish1 committed:
I didn't like that ambience

CBT copypasta

OceanFish1 committed:
Fixed some bad things now everything works as it should

Do not erp, erp is bad

OceanFish1 committed:
Immersion Ruined™

codermonkeys didn't think about mobs and shit :flushed:

OceanFish1 committed:
Revert "Immersion Ruined™"

This reverts commit 941c7d6bf7353b9892bc93f4d263dce98a32846a.

utilbird commented on code/modules/projectiles/guns/ballistic/laser_gatling.dm:
```diff
@@ -132,6 +139,24 @@
 /obj/item/gun/ballistic/minigun/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
 	if(ammo_pack)
+		if(obj_flags & EMAGGED)
+			if (cooldown < world.time)
```
```suggestion
			if(cooldown < world.time)
```

OceanFish1 replied:
i don't like that too so i killed space

OceanFish1 committed:
that one little space is obliberated

Space is no more. It was bought by nanotrasen and you have to pay your life to go to space right now.

RDS88-beeman commented on code/modules/projectiles/guns/ballistic/laser_gatling.dm:
```diff
@@ -132,6 +139,24 @@
+				else
+					to_chat(user, "The gun's heat sensor locked the trigger to prevent lens damage.")
+			else
+				to_chat(user, "You don't want your hands melt, let it cool down!.")
```
missing a "To" here, although i to do not want my hands melt

OceanFish1 replied:
will get on that later

KazooBard commented:
Looks great! I like the amount of mobs and substantial rewards. Now, if I am correct ti looks as if it were a multi-choice map where YOU choose where to go. Now, about cheesing. I cheesed a LOT of ruins back in the day, and I know a lot about the best ways to do it. It seems there is little to no way to get to the goodies without any obstacles. I see the triple/quadruple walls are the deluxe anti-cheese tool, and I like how you did somethign about it. I like all of the loot if not most is guarded by good NPC's. It looks wacky, fun and still interesting. Good job my dude, I hope cae adds this!

OceanFish1 commented:
> I hope cae adds this!

Yeah... Sure...

KazooBard commented:
> > I hope cae adds this!
> 
> Yeah... Sure...

😔

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+/obj/item/statuebust/toy/attack_self(mob/user)
+	if (cooldown < world.time)
+		cooldown = world.time + 450
+		user.visible_message("<span class='warning'>[user] activates [src].</span>", "<span class='notice'>You activate [src]!</span>", "<span class='italics'>You hear a music playing.</span>")
```
```suggestion
		user.visible_message("<span class='warning'>[user] activates \the [src].</span>", "<span class='notice'>You activate \the [src]!</span>", "<span class='italics'>You hear a music playing.</span>")
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		visible_message("<span class='danger'>[src] primes the grenade!.</span>")
+		addtimer(CALLBACK(src, .proc/prime), det_time)
+	else
+		return
```
Shouldn't be needed

utilbird replied:
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+
+/mob/living/simple_animal/hostile/syndicate/factory/boss/death(gibbed)
+	playsound(get_turf(src), 'sound/voice/borg_deathsound.ogg', 80, TRUE, 0)
+	visible_message("<span class='boldwarning'>[src] activates its self-destruct system!.</span>")
```
```suggestion
	visible_message("<span class='boldwarning'>\The [src] activates its self-destruct system!.</span>")
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+/obj/item/gun/ballistic/shotgun/lever_action/rack(mob/user = null)
+	..()
+	if(user.get_inactive_held_item() && prob(50) && chambered)
+		user.visible_message("<span class='rose'>With a single move of [user.p_their()] arm, [user] flips [src] and loads the chamber with a shell.</span>")
```
```suggestion
		user.visible_message("<span class='rose'>With a single move of [user.p_their()] arm, [user] flips \the [src] and loads the chamber with a shell.</span>")
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		var/chosen_sound = pick(possible_sounds)
+		playsound(get_turf(src), chosen_sound, 60, TRUE, 0)
+	else
+		return
```
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		icon_living = "facboss3"
+		ranged_cooldown_time = 10//even faster
+	else
+		return
```
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+	if(health <= 300)
+		icon_state = "facboss2"
+		icon_living = "facboss2"
+		ranged_cooldown_time = 20//less health - faster shooting
```
```suggestion
		ranged_cooldown_time = 20//less health - faster shooting
		return
```

OceanFish1 replied:
Didn't I change that already?

utilbird replied:
Apparently not

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		playsound(get_turf(src), chosen_sound, 80, TRUE, 0)
+		say("Target!")
+	else
+		return
```
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+										 /obj/item/ammo_casing/shotgun/breacher)
+	casingtype = pick(allowed_projectile_types)
+	..()
+	sleep(5)
+	playsound(get_turf(src), 'sound/weapons/shotgunpump.ogg', 50, 0, 0)
```
You should probably add this on a timer instead of using sleep()

OceanFish1 replied:
I know, but when I try to make a timer for playsound it never works. And I don't think it's a good idea to make a whole new proc for one sound.

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+				/obj/effect/mob_spawn/human/corpse/facboss)
+
+/mob/living/simple_animal/hostile/syndicate/factory/boss/Shoot()
+	var/allowed_projectile_types = list(/obj/item/ammo_casing/shotgun/beanbag,
```
```suggestion
	var/static/list/allowed_projectile_types = list(/obj/item/ammo_casing/shotgun/beanbag,
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+										 /obj/item/ammo_casing/shotgun/improvised,
+										 /obj/item/ammo_casing/shotgun/ion,
+										 /obj/item/ammo_casing/shotgun/laserslug,
+										 /obj/item/ammo_casing/shotgun/breacher)
```
Should this be defined in the shooting proc?

OceanFish1 replied:
I mean I was supposed to make each shot a random shell.

utilbird replied:
Yeah, but the actual list can be defined elsewhere. This works though

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+	pull_force = MOVE_FORCE_EXTREMELY_STRONG
+	hardattacks = TRUE
+	melee_damage = 50
+	speed = 5
```
higher speed is slower, just an fyi

OceanFish1 replied:
I know.

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		return
+
+/mob/living/simple_animal/hostile/zombie_suicide/proc/prime()
+	explosion(src.loc,0,2,3,flame_range = 3)
```
```suggestion
	explosion(src,0 ,2 ,3 ,flame_range = 3)
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		cooldown = world.time + 300
+		playsound(get_turf(src), chosen_sound, 50, TRUE, 0)
+	else
+		return
```
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		playsound(src, 'sound/machines/beep.ogg', 80, FALSE)
+		addtimer(CALLBACK(src, .proc/death), 200)
+	else
+		return
```
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+			return
+		if(prob(20))
+			var/chosen_sound = pick(idle_sounds)
+			playsound(src, chosen_sound, 50, FALSE)
```
```suggestion
	if(Aggro() || stat)
		return
	if(prob(20))
		var/chosen_sound = pick(idle_sounds)
		playsound(src, chosen_sound, 50, FALSE)
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		cooldown = world.time + 300
+		playsound(get_turf(src), chosen_sound, 70, TRUE, 0)
+	else
+		return
```
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+			return
+		if(prob(20))
+			var/chosen_sound = pick(idle_sounds)
+			playsound(src, chosen_sound, 50, TRUE)
```
```suggestion
	if(Aggro() || stat)
		return
	if(prob(20))
		var/chosen_sound = pick(idle_sounds)
		playsound(src, chosen_sound, 50, TRUE)
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		cooldown = world.time + 300
+		playsound(get_turf(src), chosen_sound, 70, TRUE, 0)
+	else
+		return
```
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		cooldown = world.time + 300
+		playsound(get_turf(src), chosen_sound, 70, TRUE, 0)
+	else
+		return
```
```suggestion
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+			return
+		if(prob(20))
+			var/chosen_sound = pick(idle_sounds)
+			playsound(src, chosen_sound, 50, FALSE)
```
```suggestion
	if(Aggro() || stat)
		return
	if(prob(20))
		var/chosen_sound = pick(idle_sounds)
		playsound(src, chosen_sound, 50, TRUE)
```

utilbird commented on code/modules/awaymissions/mission_code/TheFactory.dm:
```diff
@@ -0,0 +1,902 @@
+		playsound(get_turf(src), 'sound/weapons/sniper_rack.ogg', 80, TRUE)
+		say("I've got you in my scope.")
+	else
+		return
```
```suggestion
```

utilbird reviewed (changes requested):
There's a lot of repeatition in mob life() procs, might be worth making a subtype for them

A fair bit of unneeded return statements too xd

OceanFish1 committed:
some stuff elon pointed at

how do i code lol XDDDD

OceanFish1 committed:
for some reason power tools stopped existing

so

utilbird commented:
Should be good to go once the review above is addressed! :)

OceanFish1 committed:
one more thing

another one another one another one another one another one another one another one and another one

OceanFish1 committed:
dude what

i did change it wtf

Co-authored-by: TheFakeElon <59686430+TheFakeElon@users.noreply.github.com>

OceanFish1 commented:
well my bruv mate I did add that "return" because when I commited your suggestion it happened so there are now two returns in a row

OceanFish1 committed:
ayo what the fuck

I did change that brooooooooooooo

utilbird reviewed (approved):
Not sure why that happened

OceanFish1 commented:
Summon the code owner...............................

OceanFish1 commented:
@Crossedfall @ike709 So you're owning the code? [REDACTED]

Crossedfall reviewed (approved):
Config changes applied.

RDS88-beeman commented:
hey elon why'd you revert it

utilbird commented:
I didn't revert it, I started a revert PR which I didn't end up PRing.

I was going to test merge a revert because of the relatively large amount of complaints I heard about some of the map's design choices. But I didn't really have enough information as to what needed to be changed.

Ravellon commented:
The size of it makes BYOND shart itself and it tends to fail to fully load.
